Scripts/General/ModMercs.lua

```diff
@@ -0,0 +1,732 @@
+--[[
+
+Mercenary Mechanics,
+Author: Henrik Chukhran, 2022 - 2024
+
+ToDo:
+    - New Upgrade: Special
+    - New Upgrade: Magical Resistances
+    - Make sure that attacking them won't make other allies attack you
+    - Summoned monsters must be immediately hostile against your hostiles
+    - Finish/rework Merc_GetInfoString
+    
+]]
+
+-- Enums
+EMercUpgradeType    = {
+    Null            = 0,
+    HP              = 1,
+    Level           = 2,
+    Charges         = 3,
+    AC              = 4,
+    MageRes         = 5,
+    MeleeDamage     = 6,
+    RangeDamage     = 7,
+    Spell           = 8
+}
+
+-- Structs
+SMercUpgrade        = {
+    Level           = 0,
+    Price           = { 1000,2500,4000 },
+    EMercUpgradeType= EMercUpgradeType.Null
+}
+
+-- Holds mutable data shared across save files
+SMercSaveData       = {
+    NPC_ID          = 0,
+    FightsLeft      = 1,     
+    Upgrades        = { SMercUpgrade },
+    Released        = false,
+    ReleaseMap      = "",
+    Dead            = false,
+    HiredOnce       = false
+}
+
+-- Required by guildmaster
+SMercCredentials    = {
+    PriceHire       = 2000,
+    PriceReHire     = 500,
+    PriceResurrect  = 500,
+    TextAbout       = "I'm tough and rough.",
+    TextHired       = "Hired!",
+    TextGreeting    = "Greetings!",
+    TextFightTired  = "Too tired.",
+    TextSpecial     = "",
+    TextFireAttempt = "Are you sure?",
+    TextFireCancel  = "Phew, that was close!"
+}
+
+-- Holds non-mutable data for mercs
+SMerc               = {
+    StringID        = "none",
+    Name            = "Unnamed",
+    NPC_ID          = 525,
+    MonsterID       = 0,
+    Ability         = "",
+    FullHP          = { [1] = 100, [2] = 125, [3] = 150, [4] = 180 },
+    AC              = { [1] = 3, [2] = 8, [3] = 15, [4] = 20 },
+    Level           = { [1] = 5, [2] = 10, [3] = 15, [4] = 20 },
+    Attack1         = { [1] = "1D2", [2] = "2D3+1", [3] = "3D4+2", [4] = "4D4+3" },
+    Attack2         = nil,
+    FightsMax       = { [1] = 1, [2] = 2, [3] = 3, [4] = 4 },
+    Credentials     = table.copy(SMercCredentials),
+}
+
+-- Tables
+MercsDB             = {}
+MercNPCList         = {}
+
+-- Const 
+const.Mercenary     = {
+    UpgradeLimit    = 3,
+    Special         = {
+        Null        = "",
+        Undying     = "undying"
+    }
+}
+
+-- Functions
```

Comment:
I see that those functions are part of Merc API and we know it, because of `Merc_`. Global functions and name contains namespace details. It is a design decision, I'm not saying it's bad, but you will have to follow it everywhere if you want to be coherent. An alternative approach would use class and static functions:
```lua
-- Create the table for the class definition
local ExampleClassDefinition = {}

ExampleClassDefinition.staticFunction()
    print("Inside Static Function")
end

-- Define the new() function
ExampleClassDefinition.new = function()
    local self = {}
    return self
end

-- Prints "Inside Static Function"
ExampleClassDefinition.staticFunction()
```
https://paulwatt526.github.io/wattageTileEngineDocs/luaOopPrimer.html